roxie/roxiemem/roxiemem.cpp

```diff
@@ -314,6 +314,16 @@ static void initializeHeap(bool allowHugePages, bool allowTransparentHugePages,
                 heapTotalPages, (unsigned __int64) memsize, heapBase, (unsigned __int64) HEAP_ALIGNMENT_SIZE, heapBitmapSize);
 }
 
+static void adjustHeapSize(unsigned numPages)
+{
+    memsize_t bitmapSize = (numPages + UNSIGNED_BITS - 1) / UNSIGNED_BITS;
+    memsize_t totalPages = bitmapSize * UNSIGNED_BITS;
+    memsize_t memsize = totalPages * HEAP_ALIGNMENT_SIZE;
+    heapEnd = heapBase + memsize;
+    heapBitmapSize = (unsigned)bitmapSize;
+    heapTotalPages = (unsigned)totalPages;
```

Comment:
(This lines were copied from initializeHeap which checks the range to ensure overflow doesn't occur on the multiplies)